crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -0,0 +1,234 @@
+#[cfg(feature = "e2e-encryption")]
+use std::ops::Deref;
+
+use matrix_sdk_common::deserialized_responses::{
+    AmbiguityChanges, JoinedRoom, Rooms, SyncResponse,
+};
+use ruma::api::client::sync::sync_events::{v3, v4};
+#[cfg(feature = "e2e-encryption")]
+use ruma::UserId;
+
+use super::BaseClient;
+use crate::{
+    error::Result,
+    rooms::RoomType,
+    store::{ambiguity_map::AmbiguityCache, StateChanges},
+};
+
+impl BaseClient {
+    /// Process a response from a sliding sync call.
+    ///
+    /// # Arguments
+    ///
+    /// * `response` - The response that we received after a successful sliding
+    ///   sync.
+    #[tracing::instrument(skip(self, response))]
+    pub async fn process_sliding_sync(&self, response: v4::Response) -> Result<SyncResponse> {
```

Comment:
There's a lot of commented-out code here, are the FIXME's all still up-to-date?

Reply:
yes, unfortunately they are up to date. The will all go into the tracking issue for it.